### ISSUE TITLE
How do I call a REST method that accepts parameters from daprClient.InvokeMethodAsync()?

### ISSUE THREAD

**Issue by pattisapu01** [dapr/dotnet-sdk#464]
## In what area(s)?
dapr - .net SDK

> /area runtime

## Ask your question here
I need to call a "method" on a service via "InvokeMethodAsync()". This method accepts parameters. How do I pass the params to the InvokeMethodAsync() call?
sample method  signature in serviceB API
```csharp
/// <summary>
        /// Ping.
        /// </summary>
        /// <returns>Time.</returns>
        [HttpGet]
        [Route("ping/{param1}/{param2}")]
        public string Ping(string param1, string param2)
        {
            return $"{DateTime.Now.ToString()} param1 : {param1}, param2 : {param2}";
        }

//daprClient call:
var response = await _daprClient.InvokeMethodAsync<string, string>("serviceB", "ping", "{\"param1\" : \"one\", \"param2\" : \" two \"}");
// the above call does not work.. if I remove both the params and call, it works..

**Comment by vinayada1** [comment]
You can wrap the parameters inside an object and use it as mentioned in: https://github.com/dapr/dotnet-sdk/tree/master/samples/Client/DaprClient#invoking-services. Service B would have to deserialize the object and extract the parameters.

**Comment by amolenk** [comment]
With an HTTP Get request, the "method" to call is the route. You should be able to call your method using:

```c#
var response = await _daprClient.InvokeMethodAsync<string, string>("serviceB", "ping/one/two")
```

**Comment by rynowak** [comment]
Closing this because it's been answered. Please let us know if you have other questions.

**Comment by alexdresko** [comment]
> With an HTTP Get request, the "method" to call is the route. You should be able to call your method using:
> 
> ```cs
> var response = await _daprClient.InvokeMethodAsync<string, string>("serviceB", "ping/{arg1}/{arg2}")
> ```

@amolenk I spent what felt like an eon trying to figure out why I kept getting 404 when my route was set to `ping/one/two` using `InvokeMethodAsync`. I could easily hit `ping/one/two` in the browser, but never via `InvokeMethodAsync`. Then I ran a test with a simple `ping /{arg1}` route and it worked fine.  

Now, it seems like I keep shooting myself in the foot on this Dapr journey, so there's always a chance I'm doing something else wrong..  I'm just passing along my experience.
